kb/2023-08-23/using_rook_ceph_storage.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Using Rook Ceph Storage in Harvester
+description: Using Rook Ceph storage in Harvester
+slug: using_rook_ceph_storage
+authors:
+  - name: Hang Yu
+    title: Staff Software Engineer
+    url: https://github.com/futuretea
+    image_url: https://github.com/futuretea.png
+tags: [harvester, rook, ceph, csi]
+hide_table_of_contents: false
+---
+
+Harvester now allows you to install a [Container Storage Interface (CSI)](https://kubernetes-csi.github.io/docs/introduction.html) in your Harvester cluster to support and use external storage as the Virtual Machine's non-system data disk. Leverage different drivers for specific purposes, such as performance optimization or integration with existing in-house storage providers.
+
+The provisioner of the Virtual Machine (VM) image still uses Longhorn. Before v1.2.0, Harvester only supports using Longhorn to store VM data and does not support using external storage to store VM data.
+
+[Rook](https://rook.io) is an open-source cloud-native storage orchestrator providing the platform, framework, and support for Ceph storage to integrate with cloud-native environments natively.
+
+[Ceph](https://ceph.io) is a software-defined distributed storage system that provides file, block, and object storage, allowing you to deploy it in large-scale production clusters.
+
+Rook automates deployment and management of Ceph to provide self-managing, self-scaling, and self-healing storage services. The Rook operator builds on Kubernetes resources to deploy, configure, provision, scale, upgrade, and monitor Ceph.
+
+This article explains how to install, configure, and use [Rook Ceph](https://rook.io/docs/rook/v1.12/Getting-Started/intro/) as a data disk for a VM in Harvester.
+
+## Install Harvester Cluster
+
+Harvester's OS has an immutable design, which means most files in the  OS revert to their pre-configured state after a reboot.
+
+So we need to add some persistent paths used by Rook Ceph to `os.persistentStatePaths` in the [Harvester configuration](https://docs.harvesterhci.io/dev/install/harvester-configuration#ospersistent_state_paths).
+
+```yaml
+os:
+  persistentStatePaths:
+    - /var/lib/rook
+    - /var/lib/ceph
+  modules:
+    - rbd
+    - nbd
+```
+
+After the cluster is installed, refer to [How can I access the kubeconfig file of the Harvester cluster?](https://docs.harvesterhci.io/v1.1/faq#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster) to get the kubeconfig of the Harvester cluster.
+
+## Install Rook Ceph to Harvester
```

Comment:
do the following steps also work for the existing Ceph cluster? would prefer to add existing cluster integration steps as it would be more recommended instead of installing directly on the Harvester cluster.

Reply:
I will add the external ceph example

Reply:
Added